### ISSUE TITLE
${BUILD_NUM_JOBS} not defined in TF lite building script leads to crash

### ISSUE THREAD

**Issue by feranick** [tensorflow/tensorflow#63018]
### Issue type

Build/Install

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.15.0

### Custom code

No

### OS platform and distribution

Ubuntu 22.04

### Python version

3.10

### Current behavior?

in line 126 of file: `tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh`
```
cmake --build . --verbose -j ${BUILD_NUM_JOBS} -t _pywrap_tensorflow_interpreter_wrapper
```
the variable `${BUILD_NUM_JOBS}` is never defined. When compilation fo TF lite is carried out with cmake (docker, native) there is no limit in the number of processes used for compilation. This leads to a massive RAM/swap usage and in some cases when these resources run out, a crash (with logged error: `Killed signal terminated program cc1plus`).

The `${BUILD_NUM_JOBS}` variable should be set earlier in the script or replaced with a `-j 4`.

### Standalone code to reproduce the issue

```shell
PYTHON=python3 tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh native
```

**Comment by feranick** [comment]
Possible fix: In line 126 of file: `tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh`, replace:
```
cmake --build . --verbose -j ${BUILD_NUM_JOBS} -t _pywrap_tensorflow_interpreter_wrapper
```
with:
```
cmake --build . --verbose -j $(nproc) -t _pywrap_tensorflow_interpreter_wrapper
```

**Comment by mihaimaruseac** [comment]
You can also define the envvar before building:

```bash
BUILD_NUM_JOBS=4 PYTHON=python3 tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh native
```

In fact, this is the way the scripts have been designed to be run a long time ago. One setup script defines the variables and then calls the exported scripts.

**Comment by feranick** [comment]
This is all fine and well. Yet, it's completely undocumented. People will find  the following in the README.md page for tensorflow/lite:
```
PYTHON=python3 tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh native
```
and in more cases than not, their system will crash as the compiler sees a `-j` option with no actual value. At the very least, the documentation should reflect that.

**Comment by mihaimaruseac** [comment]
Yes, that's why I recommended in #63039 to define a default smallish (~2 / 4?) value in case users are not passing one, instead of pinning to `nproc`.

**Comment by feranick** [comment]
I'd be happy to push a PR with the added definition. On Feb 25, 2024 10:11 AM, Mihai Maruseac ***@***.***> wrote:
Yes, that's why I recommended in #63039 to define a default smallish (~2 / 4?) value in case users are not passing one, instead of pinning to nproc.

**Comment by mihaimaruseac** [comment]
Please CC me on the PR and I'll try to speed it up through the internal systems.

Thank you

**Comment by feranick** [comment]
Done. Thank you, @mihaimaruseac mihaimaruseac
